File: backend_Myecoprint/backend_myecoprint/models/transaction.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const User = require('./user');

const Transaction = sequelize.define('Transaction', {
  transaction_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    primaryKey: true,
    autoIncrement: true,
  },
  user_id: {
    type: DataTypes.INTEGER,
   
  },
  store_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  offer_id: {
    type: DataTypes.INTEGER,
  },
  points: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  transaction_type: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  transaction_date: {
    type: DataTypes.DATE,
    allowNull: false,
  },
}, {
  timestamps: false,
  tableName: 'transactions',
});

Transaction.belongsTo(User,{ foreignKey: 'user_id', as: 'user', onDelete: 'CASCADE' });

module.exports = Transaction;

File: Myecoprint_admin/ncej_admin/lib/data/module/transaction.dart
class Transaction {
  int storeId;
  int offerId;
  int points;
  String transactionType;
  DateTime transactionDate;

  Transaction({
    required this.storeId,
    required this.offerId,
    required this.points,
    required this.transactionType,
    required this.transactionDate,
  });

factory Transaction.fromJson(Map<String, dynamic> json) {
  return Transaction(
    storeId: json['store_id'] as int? ?? 0,
    offerId: json['offer_id'] as int? ?? 0,
    points: json['points'] as int? ?? 0,
    transactionType: json['transaction_type'] as String? ?? '',
    transactionDate: json['transaction_date'] != null
        ? DateTime.parse(json['transaction_date'].toString())
        : DateTime.now(),
  );
}
}
